inspire/utils/author_tokenizer.py

```diff
@@ -0,0 +1,233 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014, 2015 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+import re
+
+
+def scan_string_for_phrases(s):
+    """Scan a name string and output an object representing its structure.
+    @param s: the input to be lexically tagged
+    @type s: string
+    @return: dict of lexically tagged input items.
+        Sample output for the name 'Jingleheimer Schmitt, John Jacob, XVI.' is:
+        {
+            'TOKEN_TAG_LIST' : ['lastnames', 'nonlastnames', 'titles', 'raw'],
+            'lastnames'      : ['Jingleheimer', 'Schmitt'],
+            'nonlastnames'   : ['John', 'Jacob'],
+            'titles'         : ['XVI.'],
+            'raw'            : 'Jingleheimer Schmitt, John Jacob, XVI.'
+        }
+    @rtype: dict
+    """
+    split_on_re = re.compile('[\.\s-]')
```

Comment:
We should compile regular expression in the global scope, otherwise the compilation is not adding any benefit.